sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiFinalizeWritesDoFn.java

```diff
@@ -126,6 +140,12 @@ public void process(PipelineOptions pipelineOptions, @Element KV<String, String>
           rowsFinalized.inc(response.getRowCount());
 
           finalizeOperationsSucceeded.inc();
+          BigQuerySinkMetrics.FinalizeStreamCounter("ok").inc();
+          @Nullable Instant operationStartTime = c.getOperationStartTime();
+          Instant operationEndTime = Instant.now();
+          BigQuerySinkMetrics.UpdateRpcLatencyMetric(
```

Comment:
Is updating the metric here accurate for failed operations?

Reply:
This is clearer with the new method to update RpcRequests and RpcLatency metrics.